internal/controller/hubble/hubble.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2025 Illumio, Inc. All Rights Reserved.
+
+package hubble
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+
+	exp_credentials "github.com/illumio/cloud-operator/internal/tls"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/credentials"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	hubbleRelayExpectedServerName        = "ui.hubble-relay.cilium.io"
+	ciliumHubbleRelayServiceName  string = "hubble-relay"
+)
+
+var (
+	ErrCertDataMissingInSecret = errors.New("required certificate data (ca.crt, tls.crt, or tls.key) not found in secret")
+	ErrHubbleNotFound          = errors.New("hubble Relay service not found; disabling Cilium flow collection")
+	ErrNoPortsAvailable        = errors.New("hubble Relay service has no ports; disabling Cilium flow collection")
+)
+
+// getHubbleMTLSCertificatesFromSecret fetches ca.crt, tls.crt (client cert), and tls.key (client key)
+// from the specified Kubernetes secret.
+func getHubbleMTLSCertificatesFromSecret(
```

Comment:
The package is already called `hubble`. Repeating it in the function name stutters.
```suggestion
// getMTLSCertificatesFromSecret fetches ca.crt, tls.crt (client cert), and tls.key (client key)
// from the specified Kubernetes secret.
func getMTLSCertificatesFromSecret(
```